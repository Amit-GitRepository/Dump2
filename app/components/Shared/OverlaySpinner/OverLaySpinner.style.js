import {colors} from '../../../themes/constants.styles.js';

export default {
  container: {
    position: 'absolute',
    top: 0,
    bottom: 0,
    right: 0,
    left: 0,
    backgroundColor: colors.BACKDROP_COLOR
  },
  wrapperContainer: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center'
  }
};
