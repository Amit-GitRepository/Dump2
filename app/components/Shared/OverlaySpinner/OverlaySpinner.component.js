import PropTypes from 'prop-types';
import React, {Component} from 'react';
import styles from './OverLaySpinner.style';
import {ActivityIndicator} from 'react-native';
import {colors} from '../../../themes/constants.styles';
import {View} from 'react-native-animatable';

class OverlaySpinner extends Component {
  render () {
    const {showSpinner} = this.props;
    return (
      showSpinner ?
        <View style={styles.container}>
          <View style={styles.wrapperContainer} animation='zoomIn' duration={1000}>
            <ActivityIndicator color={colors.PRIMARY_BG_TEXT_CONTRAST} size='large'/>
          </View>
        </View> :
        null
    );
  }
}

OverlaySpinner.defaultProps = {
  showSpinner: 0
};

OverlaySpinner.propTypes = {
  showSpinner: PropTypes.number.isRequired
};

export default OverlaySpinner;
