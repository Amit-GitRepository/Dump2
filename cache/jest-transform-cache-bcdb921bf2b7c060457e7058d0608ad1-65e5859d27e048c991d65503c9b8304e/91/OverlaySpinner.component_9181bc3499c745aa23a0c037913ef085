Object.defineProperty(exports,"__esModule",{value:true});var cov_60aakjrih=function(){var path='/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/OverlaySpinner/OverlaySpinner.component.js',hash='130aab1494d6032c442a0ab5b99dcdccf3ee5cb9',global=new Function('return this')(),gcv='__coverage__',coverageData={path:'/Users/amitkumar/iTrueService/DevCode/iservice-app/app/components/Shared/OverlaySpinner/OverlaySpinner.component.js',statementMap:{'0':{start:{line:10,column:26},end:{line:10,column:36}},'1':{start:{line:11,column:4},end:{line:19,column:6}},'2':{start:{line:23,column:0},end:{line:25,column:2}},'3':{start:{line:27,column:0},end:{line:29,column:2}}},fnMap:{'0':{name:'(anonymous_0)',decl:{start:{line:9,column:2},end:{line:9,column:3}},loc:{start:{line:9,column:12},end:{line:20,column:3}},line:9}},branchMap:{'0':{loc:{start:{line:12,column:6},end:{line:18,column:12}},type:'cond-expr',locations:[{start:{line:13,column:8},end:{line:17,column:15}},{start:{line:18,column:8},end:{line:18,column:12}}],line:12}},s:{'0':0,'1':0,'2':0,'3':0},f:{'0':0},b:{'0':[0,0]},_coverageSchema:'332fd63041d2c1bcb487cc26dd0d5f7d97098a6c'},coverage=global[gcv]||(global[gcv]={});if(coverage[path]&&coverage[path].hash===hash){return coverage[path];}coverageData.hash=hash;return coverage[path]=coverageData;}();var _createClass=function(){function defineProperties(target,props){for(var i=0;i<props.length;i++){var descriptor=props[i];descriptor.enumerable=descriptor.enumerable||false;descriptor.configurable=true;if("value"in descriptor)descriptor.writable=true;Object.defineProperty(target,descriptor.key,descriptor);}}return function(Constructor,protoProps,staticProps){if(protoProps)defineProperties(Constructor.prototype,protoProps);if(staticProps)defineProperties(Constructor,staticProps);return Constructor;};}();var _propTypes=require('prop-types');var _propTypes2=_interopRequireDefault(_propTypes);
var _react=require('react');var _react2=_interopRequireDefault(_react);
var _OverLaySpinner=require('./OverLaySpinner.style');var _OverLaySpinner2=_interopRequireDefault(_OverLaySpinner);
var _reactNative=require('react-native');
var _constants=require('../../../themes/constants.styles');
var _reactNativeAnimatable=require('react-native-animatable');function _interopRequireDefault(obj){return obj&&obj.__esModule?obj:{default:obj};}function _classCallCheck(instance,Constructor){if(!(instance instanceof Constructor)){throw new TypeError("Cannot call a class as a function");}}function _possibleConstructorReturn(self,call){if(!self){throw new ReferenceError("this hasn't been initialised - super() hasn't been called");}return call&&(typeof call==="object"||typeof call==="function")?call:self;}function _inherits(subClass,superClass){if(typeof superClass!=="function"&&superClass!==null){throw new TypeError("Super expression must either be null or a function, not "+typeof superClass);}subClass.prototype=Object.create(superClass&&superClass.prototype,{constructor:{value:subClass,enumerable:false,writable:true,configurable:true}});if(superClass)Object.setPrototypeOf?Object.setPrototypeOf(subClass,superClass):subClass.__proto__=superClass;}var

OverlaySpinner=function(_Component){_inherits(OverlaySpinner,_Component);function OverlaySpinner(){_classCallCheck(this,OverlaySpinner);return _possibleConstructorReturn(this,(OverlaySpinner.__proto__||Object.getPrototypeOf(OverlaySpinner)).apply(this,arguments));}_createClass(OverlaySpinner,[{key:'render',value:function render()
{cov_60aakjrih.f[0]++;var _ref=(cov_60aakjrih.s[0]++,
this.props),showSpinner=_ref.showSpinner;cov_60aakjrih.s[1]++;
return(
showSpinner?(cov_60aakjrih.b[0][0]++,
_react2.default.createElement(_reactNativeAnimatable.View,{style:_OverLaySpinner2.default.container},
_react2.default.createElement(_reactNativeAnimatable.View,{style:_OverLaySpinner2.default.wrapperContainer,animation:'zoomIn',duration:1000},
_react2.default.createElement(_reactNative.ActivityIndicator,{color:_constants.colors.PRIMARY_BG_TEXT_CONTRAST,size:'large'})))):(cov_60aakjrih.b[0][1]++,


null));

}}]);return OverlaySpinner;}(_react.Component);cov_60aakjrih.s[2]++;


OverlaySpinner.defaultProps={
showSpinner:0};cov_60aakjrih.s[3]++;


OverlaySpinner.propTypes={
showSpinner:_propTypes2.default.number.isRequired};exports.default=


OverlaySpinner;