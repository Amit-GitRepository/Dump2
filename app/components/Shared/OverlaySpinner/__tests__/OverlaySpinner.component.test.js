import OverlaySpinner from '../OverlaySpinner.component';
import React from 'react';
import renderer from 'react-test-renderer';

describe('Overlay Spinner component', () => {
  it('should render spinner when showSpinner positive', () => {
    const progressBarTree = renderer.create(<OverlaySpinner showSpinner={1}/>).toJSON();
    expect(progressBarTree).toMatchSnapshot();
  });
  it('should render nothing when showSpinner positive', () => {
    const progressBarTree = renderer.create(<OverlaySpinner showSpinner={0}/>).toJSON();
    expect(progressBarTree).toMatchSnapshot();
  });
});
