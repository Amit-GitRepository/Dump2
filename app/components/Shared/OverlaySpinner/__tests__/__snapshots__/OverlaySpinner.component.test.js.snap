// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overlay Spinner component should render nothing when showSpinner positive 1`] = `null`;

exports[`Overlay Spinner component should render spinner when showSpinner positive 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(48,60,70,0.9)",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    animation="zoomIn"
    duration={1000}
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#FFFFFF"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
</View>
`;
